### ISSUE TITLE
Phase 3: Cross-encoder reranking / similarity transform

### ISSUE THREAD

**Issue by luisquintanilla** [luisquintanilla/mlnet-text-inference-custom-transforms#4]
## Overview

Add cross-encoder support for passage reranking and semantic similarity scoring. Requires extending the tokenizer for text pair input and adding a sigmoid post-processing transform.

## Architecture

```
TextTokenizerTransformer  OnnxTextModelScorerTransformer  SigmoidScorerTransformer
  (text pair mode)                                               
  [CLS] A [SEP] B [SEP]                                    sigmoid(logit)
  type: 0 0  0   1  1                                       score  [0,1]
```

### Extend `TextTokenizerOptions`

```csharp
/// <summary>
/// Optional second text input column for text-pair models (cross-encoders, QA).
/// When set, tokens from InputColumnName get token_type_ids=0 and tokens from
/// SecondInputColumnName get token_type_ids=1, separated by [SEP].
/// </summary>
public string? SecondInputColumnName { get; set; }
```

### Changes to `TextTokenizerTransformer.Tokenize()`

When `SecondInputColumnName` is set:
1. Tokenize text A: get token IDs for segment A
2. Tokenize text B: get token IDs for segment B
3. Concatenate: `[CLS] + A_tokens + [SEP] + B_tokens + [SEP]`
4. Set `token_type_ids`: 0 for A segment (including [CLS] and first [SEP]), 1 for B segment
5. Pad/truncate to `MaxTokenLength`
6. Build attention mask

**Important**: Check if `Microsoft.ML.Tokenizers.BertTokenizer` has a text-pair overload for `EncodeToIds()`. If not, manual concatenation with special tokens is needed.

### Changes to cursor/IDataView

`TokenizerDataView` and `TokenizerCursor` need to read from two columns when `SecondInputColumnName` is set.

### `SigmoidScorerOptions`

```csharp
public class SigmoidScorerOptions
{
    public string InputColumnName { get; set; } = "RawOutput";
    public string OutputColumnName { get; set; } = "Score";
}
```

### `SigmoidScorerTransformer` cursor logic

1. Read `RawOutput`  float[] of length 1 (single logit)
2. Apply sigmoid: `score = 1.0f / (1.0f + MathF.Exp(-logit))`
3. Output: single `float` score column

### `OnnxRerankerOptions`

```csharp
public class OnnxRerankerOptions
{
    public required string ModelPath { get; set; }
    public required string TokenizerPath { get; set; }
    public string QueryColumnName { get; set; } = "Query";
    public string DocumentColumnName { get; set; } = "Document";
    public string OutputColumnName { get; set; } = "Score";
    public int MaxTokenLength { get; set; } = 512;  // cross-encoders often need longer
    public int BatchSize { get; set; } = 32;
    public int? GpuDeviceId { get; set; }
    public bool FallbackToCpu { get; set; }
}
```

Chains: `TextTokenizerEstimator` (text pair mode)  `OnnxTextModelScorerEstimator`  `SigmoidScorerEstimator`

### Extension methods

```csharp
public static SigmoidScorerEstimator SigmoidScore(
    this TransformsCatalog catalog, SigmoidScorerOptions options)

public static OnnxRerankerEstimator OnnxRerank(
    this TransformsCatalog catalog, OnnxRerankerOptions options)
```

### File layout

```
Reranking/
 SigmoidScorerOptions.cs
─ SigmoidScorerEstimator.cs
 SigmoidScorerTransformer.cs
 OnnxRerankerOptions.cs
 OnnxRerankerEstimator.cs             (facade)
```

### `samples/Reranking/MsMarcoMiniLM/`
- **Model**: `cross-encoder/ms-marco-MiniLM-L-6-v2`
- **Test**: One query, 5 candidate passages. Verify ML-related passages rank higher than cooking/sports.

```powershell
Invoke-WebRequest -Uri "https://huggingface.co/cross-encoder/ms-marco-MiniLM-L-6-v2/resolve/main/onnx/model.onnx" -OutFile "models/model.onnx"
Invoke-WebRequest -Uri "https://huggingface.co/cross-encoder/ms-marco-MiniLM-L-6-v2/resolve/main/vocab.txt" -OutFile "models/vocab.txt"
```

### `samples/Reranking/BgeReranker/`
- **Model**: `BAAI/bge-reranker-base`
- **Shows**: BGE reranking with longer documents

## Verification

- [ ] Text pair tokenization produces correct `token_type_ids` (0 for query, 1 for document)
- [ ] Sigmoid scores are in [0, 1] range
- [ ] MsMarcoMiniLM sample ranks relevant passages higher than irrelevant ones
- [ ] Composable pipeline works: tokenize-pair  score  sigmoid
- [ ] Chained estimator pipeline works
- [ ] **Existing single-text tokenization is NOT broken**  `SecondInputColumnName = null` preserves current behavior
- [ ] Existing embedding samples still work

## Dependencies
- Requires Phase 1 (issue #2)
- The tokenizer extension is the main engineering work (~100-150 lines)
- The sigmoid post-processing is trivial (< 50 lines)
